web_src/js/components/DiffFileTree.vue

```diff
@@ -109,15 +108,17 @@ export default {
     updateVisibility(visible) {
       this.fileTreeIsVisible = visible;
       localStorage.setItem(LOCAL_STORAGE_KEY, this.fileTreeIsVisible);
-      this.adjustToggleButton(this.fileTreeIsVisible);
+      this.updateState(this.fileTreeIsVisible);
     },
-    adjustToggleButton(visible) {
-      const [toShow, toHide] = document.querySelectorAll('.diff-toggle-file-tree-button .icon');
-      toShow.classList.toggle('gt-hidden', visible);  // hide the toShow icon if the tree is visible
-      toHide.classList.toggle('gt-hidden', !visible); // similarly
-
-      const diffTree = document.getElementById('diff-file-tree');
-      diffTree.classList.toggle('gt-hidden', !visible);
+    updateState(visible) {
+      const btn = document.querySelector('.diff-toggle-file-tree-button');
+      const [toShow, toHide] = btn.querySelectorAll('.icon');
+      const tree = document.getElementById('diff-file-tree');
+      const newTooltip = btn.getAttribute(visible ? 'data-hide-text' : 'data-show-text');
+      btn.setAttribute('data-tooltip-content', newTooltip);
+      toggleElem(tree, visible);
```

Comment:
For  `DiffFileTree.vue`, looks like the `v-if="fileTreeIsVisible"` is not necessary? Because the visibility is toggled by the logic here as well as in script of `box.tmpl`:
```
<script>
	if (diffTreeVisible) document.getElementById('diff-file-tree').classList.remove('gt-hidden');
</script>
```

Reply:
The `<script>` only runs once on page load to immediately set the correct state after the element is rendered into the DOM. This is to prevent flashes of incorrect content.

The vue logic is still needed to achieve toggle behaviour. I think the only tiny inefficiency is that we read `localStorage` twice, but I'd argue it's probably not worth the effort trying to query current state from the DOM.

Reply:
You are right, I ignored the fact that the script is a one time running thing.